### ISSUE TITLE
reload window 'order of operations' on multiple windows [remote-ssh]  

### ISSUE THREAD

**Issue by umdstu** [microsoft/vscode-remote-release#4228]
When I have multiple windows open, each with an ssh session with the same remote container, and the remote container is shutdown, the sessions all lose connection (naturally). However, the odd behavior is that you go to the reload the windows, you can't click anything in the popup in all but one of the open windows. There appears to be a specific order the windows need to be reconnected in until the others can be reloaded. It does this consistently each morning I start up for the day.

Eg. I have 3 windows open. Window 1 and 3 would not reload until I reloaded window 2.

> Version: 1.52.1
> Commit: ea3859d4ba2f3e577a159bc91e3074c5d85c0523
> Date: 2020-12-16T16:32:10.090Z
> Electron: 9.3.5
> Chrome: 83.0.4103.122
> Node.js: 12.14.1
> V8: 8.3.110.13-electron.0
> OS: Linux x64 4.14.209-160.339.amzn2.x86_64
> Remote - SSH Extension: v0.62.0

**Comment by mayel** [comment]
I've had the same issue for a while.

**Comment by roblourens** [comment]
> However, the odd behavior is that you go to the reload the windows, you can't click anything in the popup in all but one of the open windows.

Can you explain this some more?

**Comment by umdstu** [comment]
Sure thing.

Suppose you have 4 separate vscode windows open, each a remote-ssh connection to a workspace. Then suppose for whatever reason, you lose connection to that server, severing your ssh connections. You get a blocking modal for each vscode window asking if you want to reconnect your session, or cancel (can't remember the exact wording of the options of the buttons at the moment). The issue here is that 3 of the 4 modals have non-clickable buttons, as if there were a window above them that had focus. The one that is clickable is random. Once you find the one of the 4 that works, and you re-establish the ssh connection for that window, one of the remaining 3 windows will become clickable, again a random one. After finding the one of the three that is clickable, you then have 2 remaining windows left, one that has clickable buttons and one that does not.  After you figure out which one is clickable, then you can then click the 4th and last windows buttons.

A bit wordy, I hope that helps.

Thanks

**Comment by roblourens** [comment]
That's really strange. I wonder whether it's just an issue with native modal dialogs on your OS. I can't repro on mac. 

Is this your local OS, or if not what is?

> `OS: Linux x64 4.14.209-160.339.amzn2.x86_64`

**Comment by mayel** [comment]
If so, you may want to know I'm running Manjaro with XFCE.

**Comment by roblourens** [comment]
You could try setting `"window.dialogStyle": "custom",` to avoid using the native dialogs. I would bet that is the cause.

**Comment by umdstu** [comment]
I'll let you know in the morning whether this fixes it! Thanks for the suggestion.

And yes, that is the dialog i'm seeing, but it's the styled using the native window styling. Hopefully this settings fixes it.

**Comment by mayel** [comment]
> You could try setting `"window.dialogStyle": "custom",` to avoid using the native dialogs. I would bet that is the cause.

That solves it for me :smile:

**Comment by umdstu** [comment]
I can confirm it appeared to resolve it for me as well. And yes, `OS: Linux x64 4.14.209-160.339.amzn2.x86_64` is my OS, it's [Amazon Linux 2](https://aws.amazon.com/amazon-linux-2/)

**Comment by roblourens** [comment]
Ok, this is suboptimal but I don't think there is anything we can do to customize how these native dialogs affect the window, unfortunately.
